x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/indices/flyout/steps/details/reindex_details_step.tsx

```diff
@@ -72,9 +75,25 @@ export const ReindexDetailsFlyoutStep: React.FunctionComponent<{
   const hasReindexingFailed = reindexStatus === ReindexStatus.failed;
   const correctiveAction = deprecation.correctiveAction as ReindexAction | undefined;
   const isESTransformTarget = !!correctiveAction?.transformIds?.length;
+  const isMLAnomalyIndex = Boolean(indexName?.startsWith(ML_ANOMALIES_PREFIX));
 
   const { data: nodes } = api.useLoadNodeDiskSpace();
 
+  let showEsTransformsGuidance = false;
+  let showMlAnomalyReindexingGuidance = false;
+  let showReadOnlyGuidance = false;
+  let showDefaultGuidance = false;
+
+  if (isESTransformTarget) {
+    showEsTransformsGuidance = true;
+  } else if (meta.isReadonly) {
+    showReadOnlyGuidance = true;
+  } else if (isMLAnomalyIndex) {
+    showMlAnomalyReindexingGuidance = true;
+  } else {
+    showDefaultGuidance = true;
+  }
```

Comment:
Attempted to clarify this logic... We should only be showing one of these instances of guidance at a time.